packages/mds-cache/unflatteners.ts

```diff
@@ -75,82 +79,75 @@ function parseAllDeviceStates(allDeviceStates: StringifiedAllDeviceStates): { [v
   }
 }
 
-function parseTripsEvents(tripsEventsStr: StringifiedTripsEvents): TripsEvents {
+function parseTripEvents(tripEventsStr: StringifiedTripEvents): TripEvent[] {
   try {
-    const trips: TripsEvents = {}
-    // TODO: fix awkward cast/parsing, should be unnecessary with typing
-    const tripsEvents: StringifiedTripsEvents = JSON.parse(String(tripsEventsStr))
-    Object.keys(tripsEvents).map(trip_id => {
-      trips[trip_id] = []
-      tripsEvents[trip_id].map(tripEvent => {
-        trips[trip_id].push({
-          vehicle_type: tripEvent.vehicle_type as VEHICLE_TYPE,
-          timestamp: Number(tripEvent.timestamp),
-          event_type: tripEvent.event_type as VEHICLE_EVENT,
-          event_type_reason: tripEvent.event_type_reason ? (tripEvent.event_type_reason as VEHICLE_REASON) : null,
-          annotation_version: Number(tripEvent.annotation_version),
-          annotation: tripEvent.annotation
-            ? {
-                in_bound: JSON.parse(tripEvent.annotation.in_bound),
-                areas: tripEvent.annotation.areas
-              }
-            : null,
-          gps: tripEvent.gps
-            ? {
-                lat: Number(tripEvent.gps.lat),
-                lng: Number(tripEvent.gps.lng),
-                altitude: tripEvent.gps.altitude ? Number(tripEvent.gps.altitude) : null,
-                heading: tripEvent.gps.heading ? Number(tripEvent.gps.heading) : null,
-                speed: tripEvent.gps.speed ? Number(tripEvent.gps.speed) : null,
-                accuracy: tripEvent.gps.accuracy ? Number(tripEvent.gps.accuracy) : null
-              }
-            : null,
-          service_area_id: tripEvent.service_area_id ? tripEvent.service_area_id : null
-        })
+    const result: TripEvent[] = []
+    const tripEvents: StringifiedTripEvent[] = JSON.parse(tripEventsStr)
+    tripEvents.map(tripEvent => {
```

Comment:
Could use a `map` here. FWIW, it's best-practices to use a `forEach` when you're not assigning the result of the loop directly to something, e.g.
```
items.map(blah => doSomeStuff(blah))
```
should be
```
items.forEach(blah => doSomeStuff(blah))
```
while
```
const things = items.map(blah => doSomeStuff(blah))
```
is when you'd want to use a `map`. 


For this function specifically, you probably just want to do something along the lines of
 ```
const result = tripEvents.map(tripEvent => {
  return objectYouBuiltWithTernaryStuff
})
```